tests/e2e/e2e_setup_test.go

```diff
@@ -210,15 +237,22 @@ func (s *IntegrationTestSuite) SetupSuite() {
 		}
 	}
 
-	if !s.skipUpgrade {
+	switch {
+	case !s.skipUpgrade && !s.isFork:
 		s.createPreUpgradeState()
 		s.upgrade()
 		s.runPostUpgradeTests()
+	case !s.skipUpgrade && s.isFork:
+		s.createPreUpgradeState()
+		s.upgradeFork()
+		s.runPostUpgradeTests()
+	case s.skipUpgrade:
+		s.runPostUpgradeTests()
 	}
 }
```

Comment:
```suggestion
if !s.skipUpgrade {
	s.createPreUpgradeState()

	if s.isFork {
		s.upgradeFork()
	} else {
		s.upgrade()
	}
}

s.runPostUpgradeTests()
```
```suggestion
	switch {
	case !s.skipUpgrade && !s.isFork:
		s.createPreUpgradeState()
		s.upgrade()
		s.runPostUpgradeTests()
	case !s.skipUpgrade && s.isFork:
		s.createPreUpgradeState()
		s.upgradeFork()
		s.runPostUpgradeTests()
	case s.skipUpgrade:
		s.runPostUpgradeTests()
	}
}
```

Reply:
Depends on the case. 

With only a few conditionals, the choice between if and switch depends on readability.

Here, we are repeating a lot of the same code - `s.createPreUpgradeState()` and `s.runPostUpgradeTests()`. Additionally, cases are not as simple with 2 conditions, making this logic harder to grok.

If there are many conditions, switch is preferred because it is more performant: https://stackoverflow.com/questions/767821/is-else-if-faster-than-switch-case

Reply:
The biggest reason for my comment here is that the last case (`case s.skipUpgrade`) is redundant because `s.runPostUpgradeTests()` can be applied to all cases.

Having the following also works imo:
```
	switch {
	case !s.skipUpgrade && !s.isFork:
		s.createPreUpgradeState()
		s.upgrade()
	case !s.skipUpgrade && s.isFork:
		s.createPreUpgradeState()
		s.upgradeFork()
        }
        s.runPostUpgradeTests()
```